src/accelerate/utils/modeling.py

```diff
@@ -156,11 +156,24 @@ def set_module_tensor_to_device(
             param_cls = type(module._parameters[tensor_name])
             kwargs = module._parameters[tensor_name].__dict__
             if param_cls.__name__ == "Int8Params":
+                # downcast to fp16 if any
+                if new_value.dtype == torch.float32:
+                    new_value = new_value.to(torch.float16)
                 new_value = param_cls(new_value, requires_grad=old_value.requires_grad, **kwargs).to(device)
             else:
                 new_value = param_cls(new_value, requires_grad=old_value.requires_grad).to(device)
             module._parameters[tensor_name] = new_value
 
+            if module.__class__.__name__ == "Linear8bitLt" and getattr(module.weight, "SCB", None) is None:
+                # quantize only if necessary
+                if not getattr(module.weight, "SCB", None) and isinstance(device, int):
```

Comment:
It doesn't work when device is `"cuda:0"` of `torch.device.cuda`.

Reply:
I think in this method `device` is always an int for GPUs, can you share a reproducible script?

Reply:
It look like it isn't, when using a custom `device_map`. I'll try and create a smaller script because right now I'm working with 13B model with custom wrappers.
But from what I've seen it work if and only if devices in a custom `device_map` are `int`s

Reply:
I think you want to test device.type here (after converting the device received to a torch.device)

Reply:
I believe this has been addressed in [bfd7e1a](https://github.com/huggingface/accelerate/pull/1237/commits/bfd7e1afea3cee100dcbed91ae6ffd89dee09432) (also added some tests later on)